### PR TITLE
Allow owner change of continuous aggregate

### LINKED ISSUE #1277: Cannot set owner on continuous aggregation view
**Relevant system information:**
 - OS: Google Container OS
 - PostgreSQL version (output of `postgres --version`): 11
 - TimescaleDB version (output of `\dx` in `psql`): 1;3
 - Installation method: Docker / Kubernetes

**Describe the bug**

Cannot set the owner of Continuous Aggregation view. 

**To Reproduce**
The  following query:
```
ALTER VIEW view_name OWNER TO new_owner
```
returns
```
NotSupportedError: cannot alter only SET options of a continuous aggregate
```

**Expected behavior**
Owner set

**Actual behavior**

Error returned

### LINKED ISSUE #2414: Cannot change owner of continuous aggregate
Trying to change the owner of a continuous aggregate generates an error.

```
postgres=# ALTER MATERIALIZED VIEW foo_1day OWNER TO foo;
ERROR:  cannot alter only SET options of a continuous aggregate
```

### PR DESCRIPTION
**Pull request by mkindahl**
Implements support for changing the owner of the continuous aggregate.
This will change the owner of the materialized hypertable, the
user view, the partial view, and the direct view.

Fixes #2414
Fixes #1277

### CODE REVIEW

File: src/process_utility.c [hunk 1]
```diff
@@ -2814,6 +2814,20 @@ process_altertable_start_matview(ProcessUtilityArgs *args)
 				process_altercontinuousagg_set_with(cagg, view_relid, (List *) cmd->def);
 				break;
 
+			case AT_ChangeOwner:
+				/* Change owner of view */
+				AlterTableInternal(view_relid, list_make1(cmd), false);
```
Comment (erimatnor) [inline src/process_utility.c:2819]: This seems to share a lot of code with the `AT_SetTableSpace` case. Would be good to harmonize.

I also think we need to change ownership of the other objects, the direct view, partial view, etc.

Comment (mkindahl) [reply src/process_utility.c]: Fixed.

File: src/process_utility.c [hunk 2]
```diff
@@ -2780,15 +2780,44 @@ process_altercontinuousagg_set_with(ContinuousAgg *cagg, Oid view_relid, const L
...
+/* Run an alter table command on a relation given by name */
+static void
+alter_table_by_name(Name schema_name, Name table_name, AlterTableCmd *cmd)
+{
+	alter_table_by_relation(makeRangeVar(NameStr(*schema_name), NameStr(*table_name), -1), cmd);
+}
+
+/* Alter a hypertable and do some extra processing */
+static void
+alter_table_by_id(int32 hypertable_id, AlterTableStmt *stmt, AlterTableCmd *cmd,
```
Comment (erimatnor) [inline src/process_utility.c:2800]: Nit, for clarity:

```suggestion
alter_table_by_hypertable_id(int32 hypertable_id, AlterTableStmt *stmt, AlterTableCmd *cmd,
```

or


```suggestion
alter_hypertable_by_id(int32 hypertable_id, AlterTableStmt *stmt, AlterTableCmd *cmd,
```

**Review by erimatnor — APPROVED** [review]
Approving.

A couple of thoughts, though. I am wondering what happens if you try to modify (change owner) of the internal objects directly, like the materialized hypertable or internal views.? I guess that could mess things up unless we already block such changes. Or, maybe we do not care about blocking it since people that mess with internal objects are on their own.

**Comment by mkindahl** [comment]
> Approving.
> 
> A couple of thoughts, though. I am wondering what happens if you try to modify (change owner) of the internal objects directly, like the materialized hypertable or internal views.? I guess that could mess things up unless we already block such changes. Or, maybe we do not care about blocking it since people that mess with internal objects are on their own.

Well, yes, if you start to mess with the implementation objects you're outside the "comfort zone" and we cannot guarantee any form of support if that is done, but that is not the reason for not blocking them. The reason is that blocking valid operations, even on internal objects, prevent users from fixing issues that is a result bugs that we have introduced. I think in general that we should be vary of blocking any operations since access to objects can be required to resolve situations where the system is broken.

**Review by svenklemm — APPROVED** [review]

**Pull request merged**